### PR TITLE
PROP-54

### PR DESCRIPTION
**Pull request by hubertjakubiak**
Each organization can set its own slack channel, if not set value `general` is used.

### CODE REVIEW

File: app/controllers/settings_controller.rb [hunk 1]
```diff
@@ -13,8 +14,18 @@ def apply
...
+  def save_slack_channel
+    organisation.update(organisation_attributes) if current_user.admin?
+    redirect_to action: :index
+  end
+
   private
 
+  def organisation_attributes
+    params.require(:organisation)
+          .permit(:slack_channel)
```
Comment (netguru-hound) [inline app/controllers/settings_controller.rb]: Use 2 (not 6) spaces for indenting an expression spanning multiple lines.

File: README.md [hunk 2]
```diff
@@ -75,7 +75,7 @@ Auth0 integration:
 
 Slack feature:
 
-In order to post props notifications and recieve thumbs-ups, you need to set `slack.default_channel` value (must be valid channel name, e.g. `general`) in your secrets.
+In order to post props notifications and recieve thumbs-ups, you need to set slack channel for your organisation in Settings after signing in. If not set, default slack channel is `general`.
```
Comment (lubieniebieski) [inline README.md]: shouldn't we start using `kudos` wording?

File: app/controllers/settings_controller.rb [hunk 3]
```diff
@@ -13,8 +14,18 @@ def apply
     redirect_to action: :index
   end
 
+  def save_slack_channel
+    organisation.update(organisation_attributes) if current_user.admin?
```
Comment (lubieniebieski) [inline app/controllers/settings_controller.rb]: should we check if `current_user` is part of this organization?

File: app/services/notifier/slack_notifier.rb [hunk 4]
```diff
@@ -28,12 +28,16 @@ def icon
...
+        channel: slack_channel,
         username: 'props',
         color: '#0092ca',
         icon_emoji: icon,
         as_user: false,
       }
     end
+
+    def slack_channel
+      notification.prop.organisation.slack_channel || 'general'
```
Comment (lubieniebieski) [inline app/services/notifier/slack_notifier.rb]: use `AppConfig` for storing `general` name - you will be able to use it in other places

File: app/views/settings/index.html.haml [hunk 5]
```diff
@@ -24,3 +24,12 @@
...
             Weekly
       = f.submit
+.row
+  .col-xs-12 
+    %h2 Notifications Channel
+    - if current_user.admin?
+      = form_for organisation, url: save_slack_channel_settings_path, method: :post do |f|
+        .form-group
+          = f.text_field :slack_channel
+          %p If not set, default slack channel is 'general'
```
Comment (lubieniebieski) [inline app/views/settings/index.html.haml]: read `general` from `AppConfig`

File: app/views/settings/index.html.haml [hunk 6]
```diff
@@ -24,3 +24,12 @@
...
             Weekly
       = f.submit
+.row
+  .col-xs-12
+    %h2 Notifications Channel
+    - if current_user.admin?
+      = form_for organisation, url: save_slack_channel_settings_path, method: :post do |f|
+        .form-group
+          = f.text_field :slack_channel
+          %p If not set, default slack channel is '#{AppConfig.slack.default_channel}'
```
Comment (lubieniebieski) [inline app/views/settings/index.html.haml]: how about putting that inside `<code></code>` blocks?

File: README.md [hunk 7]
```diff
@@ -75,7 +75,7 @@ Auth0 integration:
 
 Slack feature:
 
-In order to post props notifications and recieve thumbs-ups, you need to set `slack.default_channel` value (must be valid channel name, e.g. `general`) in your secrets.
+In order to post kudos notifications and recieve thumbs-ups, you need to set slack channel for your organisation in Settings after signing in. If not set, default slack channel is `general`.
```
Comment (lubieniebieski) [inline README.md]: Slack with capital `S` please

**Pull request merged**